roles/mysql/tasks/main.yml

```diff
@@ -0,0 +1,199 @@
+  when:
+    - mysql_server
+  notify:
+    - restart mysql server
+
+- name: MySQL | configure mysql client
+  community.general.ini_file:
+    dest: "{{ mysql_configuration_destination }}/{{ mysql_name }}-client.cnf"
+    section: client
+    option: socket
+    value: "{{ mysql_socket }}"
+    mode: "0644"
+  when:
+    - mysql_server
+
+- name: MySQL | start and enable mysql
+  ansible.builtin.service:
+    name: "mysql"
+    state: started
+    enabled: true
+  when:
+    - mysql_server
+
+- name: MySQL | add root my.cnf
+  ansible.builtin.template:
+    src: my.cnf.j2
+    dest: /root/.my.cnf
+    mode: "0640"
+  when:
+    - mysql_server
+
+- name: MySQL | create wildcard user
+  community.mysql.mysql_user:
+    host: "{{ item }}"
+    name: root
+    password: "{{ mysql_root_password }}"
+    login_host: "{{ mysql_host }}"
+    login_unix_socket: "{{ mysql_socket }}"
+    login_user: root
+    login_password: "{{ mysql_root_password }}"
+  loop:
+    - "{{ ansible_hostname }}"
+    - 127.0.0.1
+    - ::1
+    - localhost
+  when:
+    - mysql_server
+    - running_on_server
+
+- name: MySQL | remove Anonymous users in the database
+  community.mysql.mysql_user:
+    name: ''
+    host: "{{ item }}"
+    state: absent
+  with_items:
+    - localhost
+    - "{{ ansible_hostname }}"
+    - 127.0.0.1
+  when:
+    - mysql_server
+    - running_on_server
+
+- name: MySQL | Remove the test database
+  community.mysql.mysql_db:
+    name: test
+    state: absent
+  when:
+    - mysql_server
+    - running_on_server
+  notify:
+    - restart mysql server
+
+- name: MySQL | create databases on server
+  community.mysql.mysql_db:
+    name: "{{ item.name }}"
+    state: "{{ item.state | default('present') }}"
+    target: "{{ item.target | default(omit) }}"
+    encoding: "{{ item.encoding | default(omit) }}"
+    collation: "{{ item.collation | default(omit) }}"
+    login_host: "{{ mysql_host }}"
+    login_unix_socket: "{{ mysql_socket }}"
+    login_user: root
+    login_password: "{{ mysql_root_password }}"
+  loop: "{{ mysql_databases }}"
+  loop_control:
+    label: "{{ item.name }}"
+  when:
+    - mysql_databases is defined
+    - mysql_server
+    - running_on_server
+  no_log: true
+
+- name: MySQL | create databases
+  community.mysql.mysql_db:
+    name: "{{ item.name }}"
+    state: "{{ item.state | default('present') }}"
+    target: "{{ item.target | default(omit) }}"
+    encoding: "{{ item.encoding | default(omit) }}"
+    collation: "{{ item.collation | default(omit) }}"
+    login_unix_socket: "{{ mysql_socket }}"
+    login_host: "{{ mysql_host }}"
+    login_user: root
+    login_password: "{{ mysql_root_password }}"
+  delegate_to: "{{ mysql_host }}"
+  loop: "{{ mysql_databases }}"
+  when:
+    - mysql_databases is defined
+    - running_on_server
+
+- name: MySQL | create users on server
+  community.mysql.mysql_user:
+    name: "{{ item.name }}"
+    state: present
+    password: "{{ item.password | default(omit) }}"
```

Comment:
This would allow creating a user without a password. What is the use case for that?

Reply:
was used in debugging and forgot to remove.

Reply:
We have wanted that for testing some times, but I can not see a real reason.  I'm guessing this may have come out of the other roles.